### PR TITLE
added config upate check

### LINKED ISSUE #118: Safely re-generate valhalla.json to update new keys in new versions
We had the issue that the memory was not cleared after matrix calculations and after another ram consuming calculation the application died. I came around https://github.com/valhalla/valhalla/pull/4075, where this issue is treated, as it seems. 

Within https://github.com/valhalla/valhalla/pull/4075/commits/76b33d882cc3e6421a07d72d4e1e2f793c5ba080#diff-b3846614879a01a4589cc7be405a97326b3c79e8a9e64944d4f94a41c5d478e6 the parameters `max_reserved_labels_count_matrix` and `clear_reserved_memory_matrix` are introduced, but I am missing these within `custom_files/valhalla.json` which comes with the current 3.4.0 image.

I think these (and maybe others, I don't know) are missing within the building of the `custom_files/valhalla.json` file? So it would be nice, if this could contain the current state of all parameters, which can be used to configure valhalla's behavior. Thanks!

### PR DESCRIPTION
**Pull request by chrstnbwnkl**
Fixes #118

### CODE REVIEW

File: scripts/configure_valhalla.sh [hunk 1]
```diff
@@ -116,6 +116,32 @@ files=$(echo $files | xargs)
...
 if test -f "${CONFIG_FILE}"; then
+
+  if [[ "${update_existing_config}" == "True" ]]; then 
+    echo "INFO: Found existing valhalla.json. Updating possibly missing entries."
+
+    # create temporary default config
+    valhalla_build_config > ${TMP_CONFIG_FILE}  || exit 1
+
+    # for each path in the temp config (excluding array indices)
+    jq -r 'paths | select(map(type) | index("number") | not ) | "." + join(".")' ${TMP_CONFIG_FILE} | while read key ; do
```
Comment (chrstnbwnkl) [inline scripts/configure_valhalla.sh:127]: `paths` selects path to all leaf nodes, including array items. The output of this looks like this:

```
.meili
.meili.default
.meili.default.beta
...
```

File: tests/test.sh [hunk 2]
```diff
@@ -100,6 +100,51 @@ if [[ $(last_mod_tiles $tileset_name) != $mod_date_tiles ]]; then
...
+# should have been added back
+jq -e '.meili.default.beta' "${custom_file_folder}/valhalla.json" >/dev/null
+
+if [[ $? -eq 1 ]]; then
+  echo "valhalla.json should have been updated but wasn't."
+  exit 1
+fi
+
+line_count=$(diff -y --suppress-common-lines <(jq --sort-keys . "${custom_file_folder}/valhalla_base.json") <(jq --sort-keys . "${custom_file_folder}/valhalla.json") | wc -l)
+if [[ $line_count != "0" ]]; then
```
Comment (chrstnbwnkl) [inline tests/test.sh:119]: entry should have been added back so no diff

File: tests/test.sh [hunk 3]
```diff
@@ -100,6 +100,51 @@ if [[ $(last_mod_tiles $tileset_name) != $mod_date_tiles ]]; then
...
+
+jq -e '.meili.default.beta' "${custom_file_folder}/valhalla.json" >/dev/null
+
+if [[ $? -eq 0 ]]; then
+  echo "valhalla.json should not have been updated but was"
+  exit 1
+fi
+
+line_count=$(diff -y --suppress-common-lines <(jq --sort-keys . "${custom_file_folder}/valhalla_base.json") <(jq --sort-keys . "${custom_file_folder}/valhalla.json") | wc -l)
+if [[ $line_count != "1" ]]; then
```
Comment (chrstnbwnkl) [inline tests/test.sh:140]: entry should not have been added back so exactly one line diff

File: scripts/configure_valhalla.sh (hunk 1 above)
Comment (nilsnolde) [reply scripts/configure_valhalla.sh:127]: well it's pretty much chinese to me anyways and I'm happy I didn't have to give this a shot:)

File: README.md [hunk 4]
```diff
@@ -67,6 +67,7 @@ This image respects the following custom environment variables to be passed duri
 - `serve_tiles`: `True` starts the valhalla service. Default `True`.
 - `tileset_name`: The name of the resulting graph on disk. Very useful in case you want to build multiple datasets in the same directory. Default `valhalla_tiles`.
 - `traffic_name`: The name of the traffic.tar. Again, useful for serving mulitple traffic archives from the same directory. If empty, i.e. "", then no traffic archive will be built. Default `traffic.tar`.
+- `update_existing_config`: `True` updates missing keys in existing valhalla.json. Useful for updating stale config whiles to include newly introduced config parameters. Default `True`.
```
Comment (nilsnolde) [inline README.md:70]: this bit reads a bit strange: `Useful for updating stale config whiles to include newly introduced config parameters.`

**Review by nilsnolde — CHANGES_REQUESTED** [review]
looks good, thanks. just the readme description seems a bit borked.

File: README.md (hunk 4 above)
Comment (chrstnbwnkl) [reply README.md:70]: think I forgot to English for a second there

**Pull request merged**